### PR TITLE
Add three levels of durability

### PR DESCRIPTION
**Pull request by casey**
The existing tests pass, but I didn't add any tests for the new durability level.

Why kinds of tests would be appropriate?

### CODE REVIEW

**Comment by casey** [comment]
I'm kind of sad that I removed `commit_helper_inner`, it would have been more fun to add a third layer, `commit_helper_inner_nested`.

File: src/tree_store/page_store/mmap.rs [hunk 1]
```diff
@@ -26,8 +26,18 @@ impl Mmap {
         self.inner.len()
     }
 
-    #[cfg(target_os = "macos")]
     pub(crate) fn flush(&self) -> Result {
+        self.inner.flush()?;
```
Comment (cberner) [inline src/tree_store/page_store/mmap.rs:30]: On macos, this needs to call F_FULLFSYNC, doesn't it?

**Review by cberner — COMMENTED** [review]
I can't think of a good way to add a test for this. The ideal test would be to do a bunch of eventually durable commits, and then unplug power from the hard disk. Not exactly easy to implement though =P

File: src/tree_store/page_store/mmap.rs (hunk 1 above)
Comment (casey) [reply src/tree_store/page_store/mmap.rs:31]: Ah, right, fixed.

**Comment by casey** [comment]
Why don't we duplicate a basic test that does a bunch of commits and change it to use eventually durable commits. It won't really test durability guarantees, but it will at least exercise the codepath, and it sounds like more testing infrastructure is needed to really exercise durability guarantees.

**Comment by cberner** [comment]
Ya, that seems good for now. I'd suggest `integration_tests::persistence` that tests that data is persisted across closing & re-opening the database

**Comment by casey** [comment]
Added a test for eventual persistence, this should be ready to go.

**Comment by cberner** [comment]
Awesome, merged!

**Pull request merged**